### PR TITLE
feat: added avatar

### LINKED ISSUE #1948: 221201 - Add/Update Avatars
Screen Shot 2022-12-01 at 8 53 32 PM
Screen Shot 2022-12-01 at 8 53 24 PM

We just finished a new avatar framework and documentation. We are introducing a new "default" avatar that will be used when users fail to upload an avatar image. This new default avatar introduces 7 color combinations (background color and text color) to bring color to the experience and to get rid of the current "sad man" default avatar.

Color is randomly assigned to each user, but the color should follow the user across IOS/Android/HTML. Meaning if a user gets the orange color combo based on their username, the orange color should follow them across platforms. To help demonstrate this, @Robin Hamilton-Pennell built a quick html test page for all teams to reference and use the algorithm/setup. If you have any questions about how this works, please reach out to both her and I. The link is provided below.
Also, avatars that currently use the "sad man" default avatar should be updated with this new version.

All avatars are designed at 48px and scaled up or down proportionally from there. The single character within the default avatar (coming from the first character of the username) is using the "large 2 bold" type style (24/32) and is vertically and horizontally centered in the 48px circle.

The new "ebay" and updated "Signed out" avatars also need to be updated for both light/dark modes. They are included as 48px square assets that should be cropped in circles. Unsure if you can update these or if there is a service/other team we need to reach out to in order to make these updates. Please let us know.

Best,
Ryan

**Figma Avatar Documentation:**
https://www.figma.com/file/zEBdEhbonrBOGzZ0fXzWvM/eBay-Design-System?node-id=92847%3A108319&t=CffcRcjrh9RKUp8e-4

**Algorithm/Reference to use across platforms to ensure consistent default avatar color by username (built by @Robin Hamilton-Pennell of IOS):**
https://pages.github.corp.ebay.com/rhamiltonpennell/avatar/avatar.html

### PR DESCRIPTION
**Pull request by agliga**
Fixes #1948

- [X] This PR contains CSS changes
- [ ] This PR does not contain CSS changes

## Description
* Added avatar module. Added examples.
* I kept colors simple, like `avatar--teal`. I don't think users should have to do `avatar--teal3` or `avatar--t3`. I think it's cleaner this way.
* Added sizes based on the each of the avatar sizes. 
* I also added avatar signed out. I did not add a size to this since this is meant to be used in different sizes (as per design).
* Verify accessibility (I used the similar thing we had from badge)

## Screenshots
Screen Shot 2023-04-28 at 9 27 31 AM
Screen Shot 2023-04-28 at 9 27 39 AM

## Checklist

- [X] I verify the build is in a non-broken state
- [X] I verify all changes are within scope of the linked issue

- [X] I regenerated all CSS files under dist folder
- [X] I tested the UI in all supported browsers
- [ ] I did a visual regression check of the components impacted by doing a Percy build and approved the build
- [X] I tested the UI in dark mode and RTL mode
- [X] I added/updated/removed Storybook coverage as appropriate

### CODE REVIEW

**Review by LuLaValva — APPROVED** [review]
Looks to be in working shape to me, I like your profile names in the aria labels.

File: src/less/avatar/avatar.less [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    background-color: var(--color-marketing-teal-background-5);
+    border-radius: 48px;
+    color: var(--color-marketing-teal-foreground-5);
+    display: inline-flex;
+    font-size: @font-size-large-2;
+    font-weight: @font-weight-bold;
+    height: 48px;
+    justify-content: center;
+    overflow: hidden;
+    text-align: center;
```
Comment (ArtBlue) [inline src/less/avatar/avatar.less:14]: I think `justify-content: center;` is centering things already. I'm not seeing this as necessary. In. fact, excluding it doesn't seem to be making any difference.

File: src/less/avatar/avatar.less [hunk 2]
```diff
@@ -0,0 +1,111 @@
+@import "../variables/variables.less";
+
+.avatar {
```
Comment (ArtBlue) [inline src/less/avatar/avatar.less:3]: The letters don't look to be vertically centered. 
image

**Review by ArtBlue — COMMENTED** [review]
Looks good. Some minor things...

File: src/tokens/evo-light.css [hunk 3]
```diff
@@ -71,4 +71,18 @@
     --color-data-viz-tooltip-shadow-secondary: #0000002b;
     --color-scrim-image: rgba(0, 0, 0, 0.04);
     --color-scrim-background: rgba(0, 0, 0, 0.32);
+    --color-marketing-lime-foreground-4: var(--color-lime-7);
```
[the remark this answers is not in the document]
Comment (agliga) [reply src/tokens/evo-light.css:74]: These are global marketing variables. We should expose them at a global level.

File: src/less/avatar/avatar.less [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    object-fit: cover;
+    width: 48px;
+}
+
+.avatar--teal {
+    background-color: var(--color-marketing-teal-background-5);
+    color: var(--color-marketing-teal-foreground-5);
+}
+
+.avatar--light-teal {
```
Comment (ianmcburnie) [inline src/less/avatar/avatar.less:33]: Will this naming convention hold up if new avatars are added? For example, if an entry is added for `--color-marketing-orange-background-3` would the existing `.avatar-orange` have to be renamed `avatar--light-orange`?

Comment (agliga) [reply src/less/avatar/avatar.less:33]: I believe these are mostly in  the center. The colors we have currently are the base. If we add new ones we will call those lighter/darker. I also think if we need to, we can have a refactor to change those colors.
We can go with the `teal3`, but then it seems arbitrary.

File: src/less/avatar/avatar.less [hunk 5]
```diff
@@ -0,0 +1,111 @@
+@import "../variables/variables.less";
+
+.avatar {
+    align-items: center;
+    background-color: var(--color-marketing-teal-background-5);
+    border-radius: 48px;
```
[the remark this answers is not in the document]
Comment (ArtBlue) [reply src/less/avatar/avatar.less:6]: Can you make this change to simplify the component? The rest looks fine.

**Review by ArtBlue — CHANGES_REQUESTED** [review]

**Review by ianmcburnie — APPROVED** [review]

**Pull request merged**